### PR TITLE
Custom input ( sending first testcase with problem to test while running code )

### PR DESCRIPTION
**Pull request by somnath77**

### CODE REVIEW

File: app/controllers/api_controller.rb [hunk 1]
```diff
@@ -26,4 +26,8 @@ def serialize_resource(resources, serializer, root = nil, extra = {})
     opts = { each_serializer: serializer, root: root }.merge(extra)
     ActiveModelSerializers::SerializableResource.new(resources, opts) if resources
   end
+
+  def serialize_single_resource(resources, serializer, _root = nil, _extra = {})
```
Comment (anilmaurya) [inline app/controllers/api_controller.rb:30]: where we are using serialize_single_resource ?

Comment (somnath77) [reply app/controllers/api_controller.rb:30]: removed, it was needed when sending single problem object, now not necessary to serialize array of problems

File: app/serializers/problem_details_serializer.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class ProblemDetailsSerializer < ActiveModel::Serializer
+  attributes :id, :title, :description, :time_in_minutes, :submission_count, :test_case
+
+  def test_case
+    object.test_cases.first.input
```
Comment (anilmaurya) [inline app/serializers/problem_details_serializer.rb:7]: This will break if no test cases are added for problem

Comment (somnath77) [reply app/serializers/problem_details_serializer.rb]: returning nil if no test case added to problem

**Review by anilmaurya — APPROVED** [review]

**Pull request merged**